lib/iris/_shapefiles.py

```diff
@@ -0,0 +1,236 @@
+                            Defaults to 0.0 (eg only test intersection)
+
+    Returns
+    -------
+        A :class:`np.array` of the shape of the x & y coordinates of the cube, with points to mask equal to True
+
+    """
+    from iris.cube import Cube, CubeList
+
+    try:
+        msg = "Geometry is not a valid Shapely object"
+        if shapely.is_valid(geometry) is False:
+            raise TypeError(msg)
+    except Exception:
+        raise TypeError(msg)
+    if not isinstance(cube, Cube):
+        if isinstance(cube, CubeList):
+            msg = "Received CubeList object rather than Cube - \
+            to mask a CubeList iterate over each Cube"
+            raise TypeError(msg)
+        else:
+            msg = "Received non-Cube object where a Cube is expected"
+            raise TypeError(msg)
+    if minimum_weight > 0.0 and isinstance(
+        geometry,
+        (
+            sgeom.Point,
+            sgeom.LineString,
+            sgeom.LinearRing,
+            sgeom.MultiPoint,
+            sgeom.MultiLineString,
+        ),
+    ):
+        minimum_weight = 0.0
+        warnings.warn(
+            """Shape is of invalid type for minimum weight masking,
+            must use a Polygon rather than Line shape.\n
+              Masking based off intersection instead. """,
+            IrisDefaultingWarning,
+        )
+
+    # prepare 2D cube
+    y_name, x_name = _cube_primary_xy_coord_names(cube)
+    cube_2d = cube.slices([y_name, x_name]).next()
+    for coord in cube_2d.dim_coords:
+        if not coord.has_bounds():
+            coord.guess_bounds()
+    trans_geo = _transform_coord_system(geometry, cube_2d)
+
+    y_coord, x_coord = [cube_2d.coord(n) for n in (y_name, x_name)]
+    x_bounds = _get_mod_rebased_coord_bounds(x_coord)
+    y_bounds = _get_mod_rebased_coord_bounds(y_coord)
+    # prepare array for dark
+    box_template = [
+        sgeom.box(x[0], y[0], x[1], y[1]) for x, y in product(x_bounds, y_bounds)
+    ]
+    # shapely can do lazy evaluation of intersections if it's given a list of grid box shapes
+    # delayed lets us do it in parallel
+    intersect_template = dask.delayed(shapely.intersects(trans_geo, box_template))
+    # we want areas not under shapefile to be True (to mask)
+    intersect_template = np.invert(intersect_template).compute()
+    # now calc area overlaps if doing weighted comparison
+    if minimum_weight > 0.0:
+        for count, bol in enumerate(intersect_template):
+            if not bol:
+                intersect_area = trans_geo.intersection(box_template[count]).area
+                if (intersect_area / box_template[count].area) <= minimum_weight:
+                    intersect_template[count] = True
+                else:
+                    intersect_template[count] = False
+    mask_template = np.reshape(intersect_template, cube_2d.shape[::-1]).T
+    return mask_template
+
+
+def _template_func(bounds_array):
```

Comment:
Thanks to 162bd45a117eeeb707d54b73aeab230fb7361cc9 this function can be removed.